README.md

```diff
@@ -197,6 +203,9 @@ func ReadObj(
 
 ----
 ### 2.4. Immutable Return Values
+
+*onokonem: do we really need the immutable return values? what for?*
```

Comment:
I'm also interested in seeing explanation why immutable return value is needed.
(Especially in combination with const fields.)

Reply:
Sometimes you want to prevent something that you return to be mutated by the caller.

If, for example, you return an **internal** slice of objects from a struct method and you want to ensure the caller can't mutate neither the slice nor the objects inside to preserve encapsulation:
```
type Client interface {
  const RemoteAddress() string
  Close() error
}
type Server struct {
  connectedClients []Client
}

// GetConnectedClients is an immutable method that returns a
// deeply immutable reference to the internal slice of clients
func (s * const Server) GetConnectedClients() const [] const Client {
  return const [] const Client(s.connectedClients)
}

func main() {
  server := Server{}
  clients := server.GetConnectedClients()

  // We can read the clients
  log.Print("connected clients: ", len(clients))
  log.Print("first clients IP:", clients[0].RemoteAddress())

  // But we can't mutate them and call mutating methods
  clients[0] = nil      // Compile-time error
  clients[0].Close() // Compile-time error
}
```

Currently we'd have to:
- copy the entire slice before returning it to the outside of the struct scope to avoid nasty aliasing, which is both clunky and inefficient
- and define an immutable Client interface with 
```
func (s *Server) GetConnectedClients() []ReadOnlyClient {
  connectedClients := make([]ReadOnlyClient, len(s.connectedClients))
  for i, clt := range s.connectedClients {
    connectedClients[i] = clt.(*client)
  }
  return connectedClients
}
```